File: config/rules/opengrep/csharp/lang/cwe-321-hardcoded-jwt-key/hardcoded-jwt-key.cs
using System.Text;
using Microsoft.IdentityModel.Tokens;

public class JwtServiceExamples
{
    public void Configure(IConfiguration configuration)
    {
        // ruleid: csharp.lang.security.cwe-321.hardcoded-jwt-key
        var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes("my-super-secret-key-12345"));

        // ok: csharp.lang.security.cwe-321.hardcoded-jwt-key
        var envKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(configuration["Jwt:Key"]));
    }
}

File: config/rules/opengrep/python/lang/cwe-321-hardcoded-jwt-key/hardcoded-jwt-key.py
import os

import jwt

# ruleid: python.lang.security.cwe-321.hardcoded-jwt-key
token = jwt.encode(payload, "my-hardcoded-secret-key", algorithm="HS256")

# ok: python.lang.security.cwe-321.hardcoded-jwt-key
token = jwt.encode(payload, os.environ["JWT_SECRET"], algorithm="HS256")
